Userland/Libraries/LibGUI/Shortcut.h

```diff
@@ -8,37 +9,70 @@
 
 #include <AK/Traits.h>
 #include <Kernel/API/KeyCode.h>
+#include <LibGUI/Event.h>
 
 namespace GUI {
 
 class Shortcut {
 public:
     Shortcut() = default;
     Shortcut(u8 modifiers, KeyCode key)
-        : m_modifiers(modifiers)
-        , m_key(key)
+        : m_type(Type::Keyboard)
+        , m_modifiers(modifiers)
+        , m_keyboard_key(key)
     {
     }
     Shortcut(KeyCode key)
-        : m_modifiers(0)
-        , m_key(key)
+        : m_type(Type::Keyboard)
+        , m_modifiers(0)
+        , m_keyboard_key(key)
+    {
+    }
+    Shortcut(u8 modifiers, MouseButton button)
+        : m_type(Type::Mouse)
+        , m_modifiers(modifiers)
+        , m_mouse_button(button)
+    {
+    }
+    Shortcut(MouseButton button)
+        : m_type(Type::Mouse)
+        , m_modifiers(0)
+        , m_mouse_button(button)
     {
     }
 
-    bool is_valid() const { return m_key != KeyCode::Key_Invalid; }
-    u8 modifiers() const { return m_modifiers; }
-    KeyCode key() const { return m_key; }
+    enum class Type : u8 {
+        Keyboard,
+        Mouse,
+    };
+
     String to_string() const;
+    Type type() const { return m_type; }
+    bool is_valid() const { return m_type == Type::Keyboard ? (m_keyboard_key != Key_Invalid) : (m_mouse_button != MouseButton::None); }
+    u8 modifiers() const { return m_modifiers; }
+
+    KeyCode key() const
+    {
+        VERIFY(m_type == Type::Keyboard);
+        return m_keyboard_key;
+    }
+
+    MouseButton mouse_button() const
+    {
+        VERIFY(m_type == Type::Mouse);
+        return m_mouse_button;
+    }
 
     bool operator==(Shortcut const& other) const
     {
-        return m_modifiers == other.m_modifiers
-            && m_key == other.m_key;
+        return m_modifiers == other.m_modifiers && m_type == other.m_type && m_keyboard_key == other.m_keyboard_key && m_mouse_button == other.m_mouse_button;
     }
 
 private:
+    Type m_type { Type::Keyboard };
     u8 m_modifiers { 0 };
-    KeyCode m_key { KeyCode::Key_Invalid };
+    KeyCode m_keyboard_key { KeyCode::Key_Invalid };
+    MouseButton m_mouse_button { MouseButton::None };
```

Comment:
A very minor point: In the commit message you mention using a union, but don't actually use a union. You could use a `Variant<KeyCode, MouseButton>` here if you wanted, that's our type-safe union class, but what you've done here is fine. Just the commit message is a little confusing.

Reply:
Mm good catch, I think I originally tried a proper `union` but decided it was more trouble than it was worth but must not have remembered when writing the message. Will update the commit message 👍